### PR TITLE
Adds new tasty fish to deep fryers.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This PR introduces three new fish that can be caught from deep fryers. That's right, deep fryers are also fishing spots now. Is it silly? Yeah, but this is more or less the reason I made a whole PR to make fish edible. They've two gimmicks: one is that they're already fried, and neither raw nor gorey and can be cut into nuggets. The other is that they can evolve into the next type just by growing, no need of pairing them with each other, starting from the 'fryish', then the 'fritterish' (and its two variants: 'bernard-fish' and 'matthew-fish', big pun on the english food company) and finally the 'nessie-fish' which is very rare and big and you'd probably get 20 nuggets by cutting it alone.

Other than that, this PR adds a simple growth mechanic to fish, where they get a bit larger each time they're fed. The gained size and weight depends on several factors like hunger, their current size and weight and how much until they hit the maximum cap of twice the initial size and weight. This means bigger fish grow slowier, and it's better to feed them once in a while but not every other second. Obviously size and weight influence a lot of fish mechanics, so it's good to have a way to increase these values outside of breeding RNG (which is very shallow).
TL;DR: Feed the fish once their hunger reaches 50% circa for maximum growth.

Included in the PR are a few tidbits, like fixing a little mistake around the 'picky eater' fish trait, more readable code around the fish_growth component (unrelated to the aforementioned growth mechanic) and hunger or adding the fish evolution datum for armored pikes (normal pikes plus stinger trait, which I basically forgot to), and adjusting how weight and size of offsprings is calculated.

## Why It's Good For The Game
First and foremost, a brand new on-station fishing spot that is not the fishing portal, toilets (or the rare moisture traps in maint). Second, a fish growth mechanic through which weight and size can be increased.
Third, a few code things. Four, little bugs, now fixed.

Fifth, have a screenshot of me fucking around to find out how many bites it takes to eat all of the deepfried gargantuan nessie (53, and I had to spawn another 10 humans to finish it):
immagine

## Changelog

:cl:
add: You can now fish new, tasty treats by the station deep fryers.
add: You can now grow fish inside an aquarium by feeding them regularly (at 50% hunger for maximum growth).
add: Added the evolution for pikes to armored pikes.
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
one phillip j. fry fish please

**Comment by deathrobotpunch** [comment]
finally. Peak cooking content

**Comment by ZephyrTFA** [comment]
You have reached your yearly quota on fish prs

File: code/modules/fishing/fish/fish_evolution.dm [hunk 1]
```diff
@@ -40,24 +40,44 @@ GLOBAL_LIST_EMPTY(fishes_by_fish_evolution)
...
 /**
  * The main proc that checks whether this can happen or not.
- * Please do keep in mind a mate may not be present for fish with the
- * self-reproductive trait.
+ * Keep in mind the mate and aquarium arguments may be null if
+ * the fish is self-reproducing or this evolution is a result of a fish_growth component
  */
 /datum/fish_evolution/proc/check_conditions(obj/item/fish/source, obj/item/fish/mate, obj/structure/aquarium/aquarium)
 	SHOULD_CALL_PARENT(TRUE)
 	//chances are halved if only one parent has this evolution.
```
Comment (Jacquerel) [inline code/modules/fishing/fish/fish_evolution.dm:48]: Nitpick but you probably also want to move this comment

**Review by Jacquerel — APPROVED** [review]

File: code/modules/fishing/fish/fish_evolution.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/fishing/fish/fish_evolution.dm:48]: Where to?

Comment (Jacquerel) [reply code/modules/fishing/fish/fish_evolution.dm:48]: Just inside the if statement probably? So that it's commenting the line it describes, not the line above the line it describes.
